checks/src/main/kotlin/com/faithlife/lint/ForEachFunctionDetector.kt

```diff
@@ -10,23 +10,48 @@ import com.android.tools.lint.detector.api.Scope
 import com.android.tools.lint.detector.api.Severity
 import com.android.tools.lint.detector.api.SourceCodeScanner
 import com.intellij.psi.PsiMethod
+import org.jetbrains.kotlin.psi.KtLambdaExpression
 import org.jetbrains.uast.UCallExpression
 import org.jetbrains.uast.kotlin.KotlinUSafeQualifiedExpression
 
 class ForEachFunctionDetector : Detector(), SourceCodeScanner {
-    override fun getApplicableMethodNames(): List<String> = listOf("forEach", "forEachIndexed")
+    override fun getApplicableMethodNames(): List<String> =
+        listOf(FOR_EACH_NAME, FOR_EACH_INDEXED_NAME)
 
     override fun visitMethodCall(context: JavaContext, node: UCallExpression, method: PsiMethod) {
         if (node.uastParent is KotlinUSafeQualifiedExpression) return
 
+        val receiver = node.receiver
         val receiverTypeClass = context.evaluator.getTypeClass(node.receiverType)
 
-        if (receiverTypeClass != null && context.evaluator.inheritsFrom(receiverTypeClass, "java.lang.Iterable", false)) {
+        if (receiver != null && receiverTypeClass != null && context.evaluator.inheritsFrom(receiverTypeClass, "java.lang.Iterable", false)) {
+            val lambda = node.valueArguments.first().sourcePsi as KtLambdaExpression
+            val receiverName = receiver.asSourceString()
```

Comment:
We shouldn't use `asSourceString` like this. There's a couple of different ways to do what you're trying to do here. The best way would be to assert some UAST type here (with a cast—maybe `UNamedElement`)  and pull a name off a property. calling `asRecursiveLogString()` on UAST types in the debugger is really helpful for understanding the shape of the AST.

You can also get the text from the sourcePsi, but according to docs this can be slow.

Reply:
`asRecursiveLogString` is very helpful thanks!

Reply:
The name method didn't work because the name of function receivers is just the name of the function and not the parameters. E.g. `listOf(1, 2, 3)` has a name of `listOf`